packages/front-end/components/Radix/Select.tsx

```diff
@@ -0,0 +1,75 @@
+import { Select as RadixSelect, Text, Flex } from "@radix-ui/themes";
+import { MarginProps } from "@radix-ui/themes/dist/cjs/props";
+import HelperText from "./HelperText";
+
+type SelectProps = {
+  label: string;
+  defaultValue?: string;
+  disabled?: boolean;
+  error?: string;
+  errorLevel?: "error" | "warning";
+  value: string;
+  setValue: (value: string) => void;
+  children: React.ReactNode;
+} & MarginProps;
+
+export function Select({
+  label,
+  defaultValue,
+  disabled = false,
+  error,
+  errorLevel = "error",
+  children,
+  value,
+  setValue,
+  ...containerProps
+}: SelectProps) {
+  return (
+    <Flex direction="column" {...containerProps}>
+      <Text as="label" size="3" weight="medium">
+        {label}
+      </Text>
+      <RadixSelect.Root
+        defaultValue={defaultValue}
+        size="3"
+        disabled={disabled}
+        value={value}
+        onValueChange={setValue}
+      >
+        <RadixSelect.Trigger
+          className={error ? "error" : undefined}
+          disabled={disabled}
+          variant="surface"
+        />
+        <RadixSelect.Content variant="soft" position="popper">
+          {children}
+        </RadixSelect.Content>
+      </RadixSelect.Root>
+      {error && (
+        <HelperText status={errorLevel} mt="1">
+          {error}
+        </HelperText>
+      )}
+    </Flex>
+  );
+}
+
+export function SelectItem({
```

Comment:
This is different from our current `SelectField` where you pass in `options: (SingleValue | GroupedValue)[];`. Is that our intent? Does this provide more control over styling or something but with more overhead when creating a Select component?

Reply:
Ah honestly maybe that's a better interface. I was just following the interface we created for the `DropdownMenu` radix component. This component allows you to disable individual items but that could a part of the `SingleValue` type. Only thing I'm not sure about with the interface we have for the current `SelectField` is how we would allow for separators.

Reply:
Yeah, you'd need to pass props like `disabled` from something like `SingleValue` to the Selector component.

Maybe `GroupedValue` is a way you can handle a selector, where you ask about selectors before or after or something in the grouped value interface.

IDK, making you compose directly when you implement the `Select` also seems reasonable, but it could lead to too many custom stylings. I'd get another opinion here if you want, tbh, but this can always be changed later before we use this component more widely.

Reply:
I don't think we'd get any crazy styling because of the very basic interface I provide for the `SelectItem`, but maybe it's more verbose (?) I'm down to just merge this now just to get it in for `TagsModal` and revisit when we want to use it in more places